orc8r/cloud/go/obsidian/access/rest_middleware.go

```diff
@@ -34,83 +34,69 @@ import (
-				c,
-				http.StatusUnauthorized,
-				"Missing Client Credentials")
+		if operator == nil {
+			return makeErr(c, http.StatusUnauthorized, "missing client credentials")
 		}
 
-		// Bypass farther identity Checks for static docs GET having an
-		// operator cert should be enough
-		if urlPath := c.Path(); perm != accessprotos.AccessControl_READ || !(strings.HasPrefix(urlPath, obsidian.StaticURLPrefix)) {
+		perms := getRequestedPermissions(c, req)
+		staticReadOnly := perms == accessprotos.AccessControl_READ && strings.HasPrefix(c.Path(), obsidian.StaticURLPrefix)
+		if !staticReadOnly {
 			// Get Request's Entities' Ids
 			ids := FindRequestedIdentities(c)
 
 			// Check Operator's ACL for required entity permissions
-			ents := make([]*accessprotos.AccessControl_Entity, len(ids))
-			for i, e := range ids {
-				ents[i] = &accessprotos.AccessControl_Entity{Id: e, Permissions: perm}
+			var ents []*accessprotos.AccessControl_Entity
+			for _, id := range ids {
+				ents = append(ents, &accessprotos.AccessControl_Entity{Id: id, Permissions: perms})
 			}
-			err = accessd.CheckPermissions(oper, ents...)
+			err = accessd.CheckPermissions(operator, ents...)
 			if err != nil {
-				if _, ok := err.(errors.ClientInitError); ok {
-					return handleError(c, http.StatusServiceUnavailable, "Service Unavailable")
-				}
-				return handleError(
-					c, http.StatusForbidden, "Access Denied (%s)", err)
+				return transformErr(err, c, http.StatusForbidden, "access denied (%s)", err)
 			}
 		}
-		// all good, call next handler
+
 		if next != nil {
-			glog.V(1).Info("Access middleware successfully verified permissions. Sending request to the next middleware.")
+			glog.V(4).Info("Access middleware successfully verified permissions. Sending request to the next middleware.")
 			return next(c)
 		}
+
 		return nil
 	}
 }
 
-// Return required request permission (READ, WRITE or READ & WRITE)
-// corresponding to the request method
-func requestPermissions(c echo.Context) accessprotos.AccessControl_Permission {
-	// As a default - require ALL permissions (Read AND Write) for all
-	// 'unclassified' methods
-	perm := accessprotos.AccessControl_READ | accessprotos.AccessControl_WRITE
-	// Find out if it's READ or WRITE
-	switch c.Request().Method {
+// getRequestedPermissions returns the required request permission (READ, WRITE
+// or READ+WRITE) corresponding to the request method.
+func getRequestedPermissions(c echo.Context, req *http.Request) accessprotos.AccessControl_Permission {
+	switch req.Method {
 	case "GET", "HEAD":
-		perm = accessprotos.AccessControl_READ
+		return accessprotos.AccessControl_READ
 	case "PUT", "POST", "DELETE":
-		perm = accessprotos.AccessControl_WRITE
+		return accessprotos.AccessControl_WRITE
 	default:
-		glog.Error(LogDecorator(c)("Unclassified HTTP Method: %s", c.Request().Method))
+		glog.V(1).Info(decorate(c, "Unclassified HTTP method '%s', defaulting to read+write requested permissions", req.Method))
```

Comment:
There are only a few HTTP methods (https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), if we are getting this log often, we should add cases for the methods that we did not expect instead of masking this error.
This should remain an error and not an optional Info.

Reply:
The idea behind this change is that, by default, we should be logging errors that indicate something is wrong *with orc8r*. V(1) info is for adding visibility into when a particular request is malformed

In this case, a bad HTTP request type is a client error, not an orc8r error, so the client should be tasked with handling the error

Note that defaulting to logging orc8r+client errors is (a) what lead to obsidian having noisy error logs and (b) the more difficult problem of, by default, parsing out whether the error logs indicate the obsidian service is unhealthy or just that a particular client (e.g. soma) is misbehaving

Reply:
it could be a 1. bad client, 2. a hacking attempt, 3. orc8r not supporting a needed HTTP method which a new browser started using. 
In case #1 if the client is "ours" it'd probably ignore & mishandle the returned error & we'll have to find this out the hard way
In case #2, we wouldn't know anything until the service is either hacked or a hacker gives up
In case #3, we'd see some browser versions misbehaving & may spend some time figuring out why

I think, with all pros & cons - the real question is - do you see this kind of errors in the logs often/ever?

Reply:
Perhaps the compromise here is `glog.Info` -- log it as info but not behind V(1)